mutt/array.h

```diff
@@ -245,10 +245,9 @@
  *       the from index must not be bigger than to index.
  */
 #define ARRAY_FOREACH_FROM_TO(elem, head, from, to)                            \
-  for (size_t ARRAY_FOREACH_IDX = (from);                                      \
-       (ARRAY_FOREACH_IDX < (to)) &&                                           \
-       ((elem) = ARRAY_GET((head), ARRAY_FOREACH_IDX));                        \
-       ARRAY_FOREACH_IDX++)
+  for (size_t idx_##elem = (from);                                             \
+       (idx_##elem < (to)) && ((elem) = ARRAY_GET((head), idx_##elem));        \
```

Comment:
I agree: `elem` must be a variable name so it's going to be a single token anyway

```suggestion
       (idx_##elem < (to)) && (elem = ARRAY_GET((head), idx_##elem));        \
```